### ISSUE TITLE
Add analyzer for IDE code style analyzers in Roslyn.sln

### ISSUE THREAD

**Issue by mavasani** [dotnet/roslyn-analyzers#2508]
IDE code style analyzers have their severity controlled via code style option's notification. We have an internal helper `DiagnosticHelper.Create` in Roslyn.sln that accounts for this severity in the reported diagnostic severity. We should flag all `Diagnostic.Create` invocations in such code style analyzers that use/check code style option.
We already had 2 bugs due to this mistake. See https://github.com/dotnet/roslyn/pull/36087 and https://github.com/dotnet/roslyn/pull/36090

**Comment by Evangelink** [comment]
@mavasani Does this really requires a new analyzer? Can't you use the banned API for this specific case?

**Comment by sharwell** [comment]
Yes we should be able to use the banned API analyzer for this case.

**Comment by Evangelink** [comment]
@mavasani if there is something that needs to be done, could you give its scope?

**Comment by mavasani** [comment]
I think it might be better to change the `DiagnosticHelper.Create` overloads to explicitly accept an `IOption` input rather then `ReportDiagnostic`, so analyzers know that the intent is to use these helpers only if they support a code style option. I am not sure how easy it is to make such a change though...
